nnvm/src/top/tensor/reduce.cc

```diff
@@ -262,5 +262,62 @@ NNVM_REGISTER_BASE_REDUCE_OP(collapse_sum)
     return Array<Tensor>{ topi::collapse_sum(inputs[0], inputs[1]->shape) };
 });
 
+template<int Type>
+inline bool InferFixedType(const NodeAttrs& attrs,
+                          std::vector<int>* in_attrs,
+                          std::vector<int>* out_attrs) {
+  // Static type inference for argmax operation. Argmax return indices which
+  // should have Int32 type as shapes do.
+  CHECK_EQ(in_attrs->size(), 1U);
+  CHECK_EQ(out_attrs->size(), 1U);
+  NNVM_ASSIGN_OUTPUT_TYPE(attrs, *out_attrs, 0, static_cast<int>(Type));
+  return true;
+}
+
+NNVM_REGISTER_BASE_REDUCE_OP(argmax)
```

Comment:
`NNVM_REGISTER_BASE_REDUCE_OP` uses ReduceParam, but does argmax and argmin support the [exclude option](https://github.com/dmlc/tvm/blob/master/nnvm/include/nnvm/top/tensor.h#L227)?

Reply:
Done, please check.

Reply:
For now, it is nice to keep all reduction operator consistent, as I assume the logics are going to be the same

Reply:
@tqchen Thank you for your comments. I got it.
@grwlf Thank you for your works. Please don't mind my above comment.